drivers/max31865/max31865.c

```diff
@@ -0,0 +1,363 @@
+    spi_acquire(dev->params->spi, dev->params->cs_pin, SPI_MODE_1, SPI_CLK_5MHZ);
+
+    /* write the configuration register: */
+    spi_transfer_reg(dev->params->spi, dev->params->cs_pin, MAX31865_ADDR_CFG_W, params->cfg_byte);
+
+    /* write the high threshold register; LSB's address is MSB's address + 1 */
+    if (_temp_to_raw(dev, dev->params->temp_high_threshold, &th_code) == 0) {
+        data_tx[0] = th_code >> 8;          /* MSB */
+        data_tx[1] = th_code & 0x00FF;      /* LSB */
+        spi_transfer_regs(dev->params->spi, dev->params->cs_pin, MAX31865_ADDR_RTD_HTHRES_MSB_W,
+                          data_tx, NULL, 2);
+    }
+
+    /* write the low threshold register; LSB's address is MSB's address + 1 */
+    if (_temp_to_raw(dev, dev->params->temp_low_threshold, &th_code) == 0) {
+        data_tx[0] = th_code >> 8;          /* MSB */
+        data_tx[1] = th_code & 0x00FF;      /* LSB */
+        spi_transfer_regs(dev->params->spi, dev->params->cs_pin, MAX31865_ADDR_RTD_LTHRES_MSB_W,
+                          data_tx, NULL, 2);
+    }
+
+    spi_release(dev->params->spi);
+
+    return 0;
+}
+
+void max31865_clear_fault(max31865_t *dev, const max31865_params_t *params, uint8_t *config)
+{
+    assert(dev);
+    assert(params);
+    // assert(config);
+
+    uint8_t cfg_byte = 0;
+    uint8_t clr_byte = 0;
+
+    spi_acquire(dev->params->spi, dev->params->cs_pin, SPI_MODE_1, SPI_CLK_5MHZ);
+    cfg_byte = spi_transfer_reg(dev->params->spi, dev->params->cs_pin, MAX31865_ADDR_CFG_R, 0);
+    if (config) {
+        *config = cfg_byte;
+    }
+    clr_byte = cfg_byte | MAX31865_CFG_CLEAR_FAULT;
+    spi_transfer_reg(dev->params->spi, dev->params->cs_pin, MAX31865_ADDR_CFG_W, clr_byte);
+    spi_release(dev->params->spi);
+}
+
+int max31865_read_raw(max31865_t *dev, uint16_t *raw_data)
+{
+    assert(dev);
+    assert(raw_data);
+
+    uint8_t data_tx[3] = { 0 };     /* data sent by MCU to device on the MOSI line */
+    uint8_t data_rx[3] = { 0 };     /* data received by MCU from device on the MISO line */
+
+    data_tx[0] = MAX31865_ADDR_RTD_MSB;
+    spi_acquire(dev->params->spi, dev->params->cs_pin, SPI_MODE_1, SPI_CLK_5MHZ);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs_pin, false, data_tx, data_rx,
+                       sizeof(data_tx));
+    spi_release(dev->params->spi);
+
+    *raw_data = byteorder_bebuftohs(data_rx + 1);
+
+    /* test error bit b0: */
+    if (*raw_data & 0x0001) {
+        return -EIO;
+    }
+
+    return 0;
+}
+
+void max31865_raw_to_data(max31865_t *dev, uint32_t raw_data, max31865_data_t *data)
+{
+    assert(data);
+
+    _raw_to_temperature(dev, raw_data, &data->rtd_temperature_cdegc);
+    data->fault = MAX31865_FAULT_NO_FAULT;
+}
+
+int max31865_read(max31865_t *dev, max31865_data_t *data)
+{
+    assert(dev && data);
+
+    uint16_t raw_data;
+    if (max31865_read_raw(dev, &raw_data) == -EIO) {
+        max31865_detect_fault(dev, &data->fault);
```

Comment:
```suggestion
```
Let the application decide whether it calls `max31865_detect_fault` afterwards when getting back the `-EIO`.